src/dex-ui-components/base/ProgressBar/useProgressBar.tsx

```diff
@@ -0,0 +1,35 @@
+import { useSteps } from "chakra-ui-steps";
+import { ReactNode } from "react";
+
+export function useProgessBar(steps: ReactNode[]) {
+  const stepProps = useSteps({
+    initialStep: 0,
+  });
+  const { activeStep, setStep, nextStep, prevStep, reset } = stepProps;
+
+  const isLastStep = activeStep === steps.length - 1;
+  const isFirstStep = activeStep === 0;
+  const previousStepLabel = isFirstStep ? "Cancel" : `Back`;
+  const nextStepLabel = isLastStep ? "Send Token" : `Next`;
```

Comment:
I have also worked on a similar Progress Bar for my multsig setting screen, i will try to make this more generic by passing the steps name in props in order to use it.